libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -181,11 +159,42 @@ internal class MapboxTripSession(
         if (state == TripSessionState.STARTED) {
             return
         }
+        setNavigatorObserver()
         tripService.startService()
         startLocationUpdates()
         state = TripSessionState.STARTED
     }
 
+    private fun setNavigatorObserver() {
+        navigator.setNavigatorObserver(
+            object : NavigatorObserver() {
+                override fun onStatus(origin: NavigationStatusOrigin, status: NavigationStatus) {
+                    mainJobController.scope.launch {
+                        val tripStatus = MapboxNativeNavigatorImpl.generateTripStatusFrom(status)
```

Comment:
How long does it take to run this function? Curious because our previous implementation required if is active blocks
``` kotlin
if (!isActive) {
    return@launch
}
```

Those were in place to prevent downstream listeners to be called after the sdk is cleaned up. There may still be a race-condition here because the `onStatus` is called before `setNavigatorObserver(null)`

Reply:
OH wait https://github.com/mapbox/mapbox-navigation-android/pull/4354/files#r627642971

We won't need a is active check after the generateTripStatusFrom - because it doesn't need to be a suspending function.

But we may need an if (!isActive) after each update* function that is notifying listeners 🤔